backend/src/main/java/de/qaware/mercury/rest/image/ShopImageController.java

```diff
@@ -46,15 +49,22 @@ public ImageDto uploadImageForShop(
         @RequestParam("file") MultipartFile file, HttpServletRequest servletRequest
     ) throws LoginException, IOException {
         Shop shop = authenticationHelper.authenticateShop(servletRequest);
+        String color = null;
 
         // Store the image
         Image image;
         try (InputStream stream = file.getInputStream()) {
             image = imageService.addImage(shop.getId(), stream);
+            try {
+                color = imageService.getImageBackgroundColor(image.getId());
+            } catch (ImageNotFoundException e) {
+                log.error("Could not derive image color. Will use default.", e);
```

Comment:
Das kann ja eigentlich nicht passieren, oder? Da könntest nen `AssertionError` schmeißen.

Reply:
Die Überlegung war halt, wenn da doch irgendwas schiefgegangen ist, dann sollte das zwar sichtbar werden, aber nicht wegen der Colorberechnung den Request brechen. Hab es nun trotzdem durch einen AssertionError ersetzt, da es ja in der Tat nicht vorkommen sollte.